iree/turbine/kernel/wave/utils.py

```diff
@@ -234,6 +250,19 @@ def is_mma(node):
             custom.vector_shapes.update(hardware_constraint.vector_shapes)
         custom.anchor = custom
         custom.reduction_dim = k
+        if last_vector_shapes and last_vector_shapes != custom.vector_shapes:
```

Comment:
Can't we check the the "vector_shapes" per operand of the MMA, as opposed to checking what is the latest vector shape we come across?

Reply:
Yes, currently this is limited to only checking the current and previous mma. Let me try and modify this to account for that.

Reply:
Modified to handle this.